Problem: bigint in POST message makes GET messages.json return a 500

A POST message was accepted containing a big integer. We should reject messages containing integers bigger than 2^53 (limit for valid JSON) as this breaks JSON serialization in the API.

```
[2023-09-13 10:14:26 +0000] [106] [ERROR] Error handling request
OverflowError: int too big to convert

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/sentry_sdk/integrations/aiohttp.py", line 126, in sentry_app_handle
    reraise(*_capture_exception(hub))
  File "/opt/venv/lib/python3.11/site-packages/sentry_sdk/_compat.py", line 60, in reraise
    raise value
  File "/opt/venv/lib/python3.11/site-packages/sentry_sdk/integrations/aiohttp.py", line 116, in sentry_app_handle
    response = await old_handle(self, request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/pyaleph/src/aleph/web/controllers/messages.py", line 231, in view_messages_list
    return format_response(
           ^^^^^^^^^^^^^^^^
  File "/opt/pyaleph/src/aleph/web/controllers/messages.py", line 203, in format_response
    return web.json_response(text=aleph_json.dumps(response).decode("utf-8"))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/pyaleph/src/aleph/toolkit/json.py", line 37, in dumps
    return orjson.dumps(obj)
           ^^^^^^^^^^^^^^^^^
TypeError: Integer exceeds 64-bit range
```

[The problematic message](https://github.com/aleph-im/pyaleph/files/12597048/message_with_bigint.txt). See `"random_number": 65810885944677670000000000000000000000000000000000000000000000000000000000000`.

--- a/src/aleph/toolkit/json.py
+++ b/src/aleph/toolkit/json.py
@@ -4,9 +4,11 @@ Makes swapping between JSON implementations easier.
 """
 
 import json
+from datetime import date, datetime, time
 from typing import IO, Any, Union
 
 import orjson
+from pydantic.json import pydantic_encoder
 
 # The actual type of serialized JSON as returned by the JSON serializer.
 SerializedJson = bytes
@@ -37,8 +39,22 @@ def dump(fp: IO, obj: Any) -> None:
     raise NotImplementedError("orjson does not provide dump")
 
 
+def extended_json_encoder(obj: Any) -> Any:
+    """
+    Extended JSON encoder for dumping objects that contain pydantic models and datetime objects.
+    """
+    if isinstance(obj, datetime):
+        return obj.timestamp()
+    elif isinstance(obj, date):
+        return obj.toordinal()
+    elif isinstance(obj, time):
+        return obj.hour * 3600 + obj.minute * 60 + obj.second + obj.microsecond / 1e6
+    else:
+        return pydantic_encoder(obj)
+
+
 def dumps(obj: Any) -> bytes:
     try:
         return orjson.dumps(obj)
     except TypeError:
-        return json.dumps(obj).encode()
+        return json.dumps(obj, default=extended_json_encoder).encode()

